streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -128,9 +134,8 @@ public boolean isOpen() {
     @Override
     public synchronized V get(final K key) {
         validateStoreOpen();
-        if (key == null) {
-            return null;
-        }
+        validateKeyNotNull(key);
```

Comment:
Why throw NPE if the key is null in `get` or `delete` - i think we can just ignore null keys